master/internal/command/command.go

```diff
@@ -57,6 +58,45 @@ func createGenericCommandActor(
 		serviceAddress: &serviceAddress,
 	}
 
+	if spec.Config.IdleTimeout != nil {
+		var getLastActivity func(ctx *actor.Context) *time.Time
+		switch {
+		case spec.WatchProxyIdleTimeout && spec.WatchRunnerIdleTimeout:
+			getLastActivity = func(ctx *actor.Context) *time.Time {
+				proxyRef := ctx.Self().System().Get(actor.Addr("proxy"))
+				services := ctx.Ask(proxyRef, proxy.GetSummary{}).Get().(map[string]proxy.Service)
+				service, ok := services[string(taskID)]
+				if !ok {
+					return nil
+				}
+
+				if cmd.lastActivity != nil && cmd.lastActivity.After(service.LastRequested) {
```

Comment:
I should've caught this in review @shiyuann but here we have unprotected concurrent access of cmd.lastActivity since it's captured by closure. We're very unlikely to hit it, but i'll fix it in the refactor I'm doing right now for task unif.

Reply:
oh wait, it's not a different actor, nvm. i forgot it just has things that look like actor methods.